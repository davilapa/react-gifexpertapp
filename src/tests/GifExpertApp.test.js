import GifExpertApp from "../GifExpertApp";
import { shallow } from 'enzyme';

describe('GifExpertApp test', () => {
    test('should render correctly', () => { 
        const wrapper = shallow(<GifExpertApp />)

        expect(wrapper).toMatchSnapshot();
     });

    test('should render a list of categories', () => { 
        const categories = [ 'One Puch Man', ' Dragon Ball Z'];

        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
     });
})