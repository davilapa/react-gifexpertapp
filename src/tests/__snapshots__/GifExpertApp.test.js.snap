// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp test should render a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Puch Man"
      id="One Puch Man"
    />
    <GifGrid
      category=" Dragon Ball Z"
      id=" Dragon Ball Z"
    />
  </ol>
</Fragment>
`;

exports[`GifExpertApp test should render correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
